src/main/java/org/jenkinsci/plugins/workflow/steps/EnvironmentExpander.java

```diff
@@ -48,6 +52,15 @@
      */
     public abstract void expand(@Nonnull EnvVars env) throws IOException, InterruptedException;
 
+    /**
+     * Override this method to allow tracking of sensitive environment variables
+     *
+     * @return the set of sensitive environment variable names to track.
+     */
+    public Set<String> getSensitiveVars() {
+        return Collections.emptySet();
+    }
```

Comment:
I would add an `Nonnull` annotation to the return value, tweak the Javadoc a bit, and consider changing the name to `getSensitiveVariables` (renaming is not critical, but I would prefer to avoid abbreviations and feel like that name would match `AbstractBuild.getSensitiveBuildVariables` better)

```suggestion
    /**
     * Get the names of environment variables known to contain sensitive values, such as secrets, in the current context.
     * Should be overridden by subclasses that bind secret values to environment variables.
     *
     * @return a set of environment variables known to contain sensitive values
     */
    public @Nonnull Set<String> getSensitiveVariables() {
        return Collections.emptySet();
    }
```